src/commentParserToESTree.js

```diff
@@ -363,15 +373,26 @@ const commentParserToESTree = (jsdoc, mode, {
         : rawType;
     }
 
-    if (description) {
+    // In `compact` mode skip processing if `description` is an empty string.
+    // In `preserve` mode process when `description` is not the `empty string
+    // or the `delimiter` is not `/**` ensuring empty lines are preserved.
+    if ((spacing === 'compact' && description) ||
+        (spacing === 'preserve' && (description || delimiter !== '/**'))) {
       const holder = lastTag || ast;
+
+      // Check if there are any description lines and if not then this is a
+      // multi-line comment block with description on 0th line. Treat
+      // `delimiter` / `postDelimiter` / `initial` as being on a new line.
+      const isDelimiterLine = holder.descriptionLines.length === 0 &&
+        delimiter === '/**';
```

Comment:
Is `holder.descriptionLines.length` ever non-zero when the delimiter is `/**`?

Reply:
It is necessary in the single line / 0th end tag section (line 266) where the following test will fail for comment block [4]:

```js
commentBlocks.push(`/**
 * Should parse one line. */`);

// A multi-line comment closing delimiter on last line creates one more line.
// Use `${' '}` to prevent IDEs from removing trailing spaces.
commentResults.push(`/**
 * Should parse one line.${' '}
 */`);
 ```
 
 Without `&& delimiter === '/**'` on line 266 the output is:
 ```js
 /**
   Should parse one line.
  */
  ```
  
In the multi-line tests none of the current tests are affected on line 366 without `&& delimiter === '/**`, but to keep parity w/ line 266 I duplicate the `isDelimiterLine` test. From the investigation I did leading new lines before a comment block are stripped, so the 0th line should always have `/**` as the delimiter.

Reply:
But you are correct that `isDelimiterLine` can just check `const isDelimiterLine = delimiter === '/**'` and no tests fail. So that could be cleaned up dropping the the `holder.descriptionLine.length === 0` check at least with the current tests. If there are more test variations for comment blocks that you can think of it is relatively easy to add them.

Reply:
> But you are correct that `isDelimiterLine` can just check `const isDelimiterLine = delimiter === '/**'` and no tests fail. So that could be cleaned up dropping the the `holder.descriptionLine.length === 0` check at least with the current tests.

So will you drop the `holder.descriptionLine.length === 0 ` then?